fs: Add FSTAB_ENTRY_DT_*_MOUNT_POINT macros

rruuaanng opened the pull request:
Since `mount-point` are commonly used in applications, I suggest adding a dedicated wrapper macro.

rruuaanng committed:
fs: Add a FSTAB_ENTRY_DT_*_MOUNT_POINT macros

These macros provide a way to get the `mount-point` value from
an fstab entry:
- FSTAB_ENTRY_DT_MOUNT_POINT
- FSTAB_ENTRY_DT_INST_MOUNT_POINT

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
subsystem: fs: Use better wrapper macros

Replace `DT_INST_PROP(inst, mount_point)` with
`FSTAB_ENTRY_DT_INST_MOUNT_POINT(inst)`.

Signed-off-by: James Roy <rruuaanng@outlook.com>

de-nordic reviewed (changes requested):
Addition looks ok; test case missing from in fs api tests.

rruuaanng committed:
tests: subsys: fs: Add FSTAB_ENTRY_DT_*_MOUNT_POINT unit test

Add unit tests for `FSTAB_ENTRY_DT_MOUNT_POINT`
and `FSTAB_ENTRY_DT_INST_MOUNT_POINT` macros.

Signed-off-by: James Roy <rruuaanng@outlook.com>
